### PR TITLE
[backend] fix max shareable markings migration (#7392)

### LINKED ISSUE #7392: Error at max shareable marking migration if version < 6.0
## Description

If upgrading from OpenCTI 6.0 to OpenCTI 6.1.11, the migration does not work.
You need to upgrade to 6.1.10 and then to 6.1.11.

## Error message

image

### PR DESCRIPTION
**Pull request by Archidoit**
PR content:
- fix issue: https://github.com/OpenCTI-Platform/opencti/issues/7392
- delete the platform_max_shareable_markings field

### CODE REVIEW

**Review by marieflorescontact — APPROVED** [review]

File: opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js [hunk 1]
```diff
@@ -48,27 +48,30 @@ export const up = async (next) => {
...
+      params: { fieldToRemove: 'platform_data_sharing_max_markings' },
+      source: 'ctx._source.remove(params.fieldToRemove)',
     },
     query: {
       bool: {
         must: [
           { term: { 'entity_type.keyword': { value: 'Settings' } } },
         ],
+        should: [
+          { exists: { field: 'platform_data_sharing_max_markings' } },
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js]: I guess this query works already, but I wonder why not add 
{ exists: { field: 'platform_data_sharing_max_markings' } } directly in "must" list just above like this:

```
query: {
      bool: {
        must: [
          { term: { 'entity_type.keyword': { value: 'Settings' } } },
          { exists: { field: 'platform_data_sharing_max_markings' } },
        ],
       },
    },
```

Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js]: yes you're right ! fixed :)

**Review by SouadHadjiat — CHANGES_REQUESTED** [review]
The migration doesn't work for me locally, the field is still present in my Settings with null, exists condition doesn't work in this case : https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-exists-query.html exists will not match null or [] values.

=> I think that you should just remove the field, without the exists condition, just the term query on entity_type "Settings"

File: opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js [hunk 2]
```diff
@@ -48,27 +48,27 @@ export const up = async (next) => {
...
-      params: { null: null },
-      source: 'ctx._source.platform_data_sharing_max_markings = params.null',
+      params: { fieldToRemove: 'platform_data_sharing_max_markings' },
+      source: 'ctx._source.remove(params.fieldToRemove)',
     },
     query: {
       bool: {
         must: [
           { term: { 'entity_type.keyword': { value: 'Settings' } } },
+          { exists: { field: 'platform_data_sharing_max_markings' } },
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js]: exists condition doesn't match null or [] values, so I think it should be removed

File: opencti-platform/opencti-graphql/src/migrations/1718631740774-remove-platform-max-shareable-markings.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const updateQuery = {
+    script: {
+      params: { fieldToRemove: 'platform_data_sharing_max_markings' },
+      source: 'ctx._source.remove(params.fieldToRemove)',
+    },
+    query: {
+      bool: {
+        must: [
+          { term: { 'entity_type.keyword': { value: 'Settings' } } },
+          { exists: { field: 'platform_data_sharing_max_markings' } },
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/migrations/1718631740774-remove-platform-max-shareable-markings.js]: same here: remove exists condition

File: opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js (hunk 2 above)
Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/migrations/1717060854132-max-shareable-marking-in-groups.js]: I'm gonna modify the condition with: field exist OR is field is null

File: opencti-platform/opencti-graphql/src/migrations/1718631740774-remove-platform-max-shareable-markings.js (hunk 3 above)
Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/migrations/1718631740774-remove-platform-max-shareable-markings.js]: done

**Review by SouadHadjiat — APPROVED** [review]

**Comment by SouadHadjiat** [comment]
make sure before merging that the new migration has an up to date timestamp compared to latest migrations on master

**Pull request merged**